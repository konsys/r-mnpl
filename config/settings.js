module.exports.settings = {
  apiHost: "localhost:3001"
};
