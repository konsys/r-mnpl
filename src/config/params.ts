export enum Params {
  BASE_URL = "http://localhost:3001",
}
