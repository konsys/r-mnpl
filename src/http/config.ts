export const configs = {
  production: {
    baseURL: window.location.origin,
    versionHash: "%VERSION_HASH%",
    versionBuild: "%VERSION_BUILD%",
    buildTime: "%BUILD_TIME%"
  },
  development: {
    baseURL: "http://localhost:3001",
    versionHash: "%VERSION_HASH%",
    versionBuild: "%VERSION_BUILD%",
    buildTime: "%BUILD_TIME%"
  }
};

// @ts-ignore
export const config = configs[process.env.NODE_ENV];

export function logVersion() {
  const styles = [
    "font-size: 15px; font-weight: bold",
    "font-size: 15px; color: #007aff"
  ];

  console.log(`%c BUILD: %c${config.versionBuild}`, styles[0], styles[1]);
  console.log(`%c HASH:  %c${config.versionHash}`, styles[0], styles[1]);
  console.log(`%c TIME:  %c${config.buildTime}`, styles[0], styles[1]);
}
