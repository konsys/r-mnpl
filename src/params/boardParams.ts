export const BOARD_PARAMS = {
  LINE_TRANSITION_TIMEOUT: 700,
  CORNER_FIELDS: [0, 10, 20, 30],
  TABLE_SIZE: 665,
  MARGIN_CENTER: 35,
  FIELD_WIDTH: 55,
  DURATION: 100,
  BANK_USER_ID: 0,
};
