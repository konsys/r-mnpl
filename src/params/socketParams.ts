export const SOCKET_PARAMS = {
    GAME_SOCKET:'http://localhost:8001/gameChat',
    ROOMS_SOCKET:'http://localhost:8002/rooms'
}