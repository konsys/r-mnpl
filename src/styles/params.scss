.table-body-board-fields-one[mnpl-group] .table-body-board-fields-one-body {
  cursor: pointer;
}

.table-body-board-fields-one[mnpl-line="0"] .table-body-board-fields-one-body ._logo,
.table-body-board-fields-one[mnpl-line="2"] .table-body-board-fields-one-body ._logo {
  transform: rotate(-90deg);
}

.table-body-board-fields-one[mnpl-line="1"] .table-body-board-fields-one-mortgaged,
.table-body-board-fields-one[mnpl-line="3"] .table-body-board-fields-one-mortgaged {
  top: 0;
}

.table-body-board-fields-one[mnpl-line="1"] .table-body-board-fields-one-level,
.table-body-board-fields-one[mnpl-line="3"] .table-body-board-fields-one-level {
  flex-direction: column;
}

.table-body-board-fields-one[mnpl-line="0"] .table-body-board-fields-one-level {
  bottom: 2px;
}

.table-body-board-fields-one[mnpl-line="0"] .table-body-board-fields-one-mortgaged {
  bottom: -7px;
}

.table-body-board-fields-one[mnpl-line="1"] .table-body-board-fields-one-label > div {
  transform: rotate(90deg);
}

.table-body-board-fields-one[mnpl-line="1"] .table-body-board-fields-one-level {
  left: 1px;
}

.table-body-board-fields-one[mnpl-line="1"] .table-body-board-fields-one-mortgaged {
  left: 2px;
}

.table-body-board-fields-one[mnpl-line="2"] .table-body-board-fields-one-level {
  top: 0;
}

.table-body-board-fields-one[mnpl-line="2"] .table-body-board-fields-one-mortgaged {
  top: -7px;
}

.table-body-board-fields-one[mnpl-line="3"] .table-body-board-fields-one-label > div {
  transform: rotate(-90deg);
}

.table-body-board-fields-one[mnpl-line="3"] .table-body-board-fields-one-level {
  right: 1px;
}

.table-body-board-fields-one[mnpl-line="3"] .table-body-board-fields-one-mortgaged {
  right: -2px;
}

.table-body-board-fields-one[mnpl-mortgaged="1"] .table-body-board-fields-one-body {
  position: relative;
  overflow: hidden;
}

.table-body-board-fields-one[mnpl-mortgaged="1"] .table-body-board-fields-one-body:after {
  content: " ";
  display: block;
  position: absolute;
  top: -99px;
  left: -99px;
  width: 999px;
  height: 999px;
  background: rgba(0, 0, 0, 0.25);
}

.table-body-board-fields._mode_choose_field .table-body-board-fields-one:not(._mode_choose_field_available) {
  cursor: default !important;
  opacity: 0.25;
}

.table-body-board-fields._mode_choose_field .table-body-board-fields-one._mode_choose_field_available {
  cursor: pointer !important;
}
body[mnpl-touchscreen="1"] .table-body-board-chatbottom > div._input ._hint._hint_tab {
  left: -42px;
}

body[mnpl-touchscreen="1"] .table-body-board-chatbottom > div._input ._hint._hint_tab span:before {
  content: "РљР»РёРєРЅРёС‚Рµ СЃСЋРґР°";
}

body:not([mnpl-touchscreen="1"]) .table-body-board-chatbottom > div._input ._hint._hint_tab span:before {
  content: "РќР°Р¶РјРёС‚Рµ Tab";
}
