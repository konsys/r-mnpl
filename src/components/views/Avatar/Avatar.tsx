import React from "react";
import { Params } from "../../../config/params";

export interface AvatarProps {
  money: number;
  remainTime: number;
  isVip: boolean;
  avatar: string;
  name: string;
}
export const Avatar = (props: AvatarProps) => (
  <div className="table-body-players-card-body">
    <div className="table-body-players-card-body-avatar">
      <div>
        <div
          style={{ backgroundImage: `url(${Params.BASE_URL}${props.avatar})` }}
        />
      </div>
    </div>
    <div className="table-body-players-card-body-nick">
      {props.isVip && <div className="_vip" />}
      <div className="_nick">
        <div>{props.name}</div>
      </div>
      <div className="_muted" style={{ display: "none" }} />
      <div className="_ignore" style={{ display: "none" }} />
    </div>
    <div className="table-body-players-card-body-money">{props.money}</div>
    <div
      className="table-body-players-card-body-timer"
      style={{ display: "block" }}
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 100 100"
        preserveAspectRatio="none"
      >
        <circle cx="50" cy="50" r="46" transform="rotate(-90 50 50)" />
      </svg>
      <div>{props.remainTime}</div>
    </div>
  </div>
);
