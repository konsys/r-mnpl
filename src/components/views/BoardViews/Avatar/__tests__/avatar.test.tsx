import { Avatar } from "../Avatar";
import { Params } from "../../../../../config/params";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Test Avatar view", () => {
  const avatar = (
    <Avatar
      money={1}
      remainTime={2}
      isVip={false}
      avatar={"testAvatareSrc"}
      name={"Ivan"}
    />
  );
  it("renders correctly", () => {
    const tree = renderer.create(avatar).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shows name", () => {
    expect(shallow(avatar).contains("Ivan")).toBeTruthy();
  });

  it("shows img src", () => {
    const wrapper = shallow(avatar);
    expect(
      wrapper.find(".table-body-players-card-body-avatar-background")
    ).toHaveLength(1);
  });
});
