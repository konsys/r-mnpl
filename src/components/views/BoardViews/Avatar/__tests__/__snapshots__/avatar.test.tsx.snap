// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Avatar view renders correctly 1`] = `
<div
  className="table-body-players-card-body"
>
  <div
    className="table-body-players-card-body-avatar"
  >
    <div>
      <div
        className="table-body-players-card-body-avatar-background"
        style={
          Object {
            "backgroundImage": "url(http://localhost:3001/testAvatareSrc)",
          }
        }
      />
    </div>
  </div>
  <div
    className="table-body-players-card-body-nick"
  >
    <div
      className="_nick"
    >
      <div>
        Ivan
      </div>
    </div>
    <div
      className="_muted"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="_ignore"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
  <div
    className="table-body-players-card-body-money"
  >
    1
  </div>
  <div
    className="table-body-players-card-body-timer"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <svg
      preserveAspectRatio="none"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        r="46"
        transform="rotate(-90 50 50)"
      />
    </svg>
    <div>
      2
    </div>
  </div>
</div>
`;
